### PR TITLE
Isolate the renderer process

### PR DESCRIPTION
**Pull request by maxcnunes**
In order to follow electron best security practices we should not use remote anymore in the renderer process. On its place we will use direct IPC calls specifically for what we need.
Sorry, it turned out to be a bit long PR, but it is doing a single thing: moved any electron/remote api usage from renderer process to the browser process and it is using ipc calls for that, but even the ipc calls aren't done directly through `ipcRenderer` module, instead every `ipcRenderer` is encapsulated by a Facade making it available to the frontend as a module.

* `src/renderer/api.ts`: is the API for every backend/browser integration
* `src/browser/preload.ts`: is where that API integrates with the backend/browser by ipcRenderer events
* `src/browser/ipcMain.ts`: is where those ipcRenderer events are received by the backend and proxied to the right module
* `src/common/types/api.ts`: has the interface `SqlectronAPI` for that integration, both renderer and process implement that to ensure the contract is properly followed and avoid any wrong implementation by them

### CODE REVIEW

File: src/browser/core/db.ts [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+  setSelectLimit(limit: number): void {
+    return setSelectLimit(limit);
+  }
+}
+
+// Keep connections by BrowserWindow instances
+const connections: {
+  [windowId: string]: SqlectronDB;
+} = {};
+
```
Comment (maxcnunes) [inline src/browser/core/db.ts:246]: Because now the db connection lives in the browser process, it means when we there are multiple application windows open they share the same browser process, and to allow each window to have its own db connection, they are stored by windowId in this object.

File: src/browser/window.ts [hunk 2]
```diff
@@ -39,7 +41,7 @@ export function buildNewWindow(app: App): void {
   // and load the index.html of the app.
   let entryBasePath = 'file://' + resolve(__dirname, '..');
   if (devMode) {
-    entryBasePath = 'http://localhost:8080';
+    entryBasePath = 'https://localhost:8080';
```
Comment (MasterOdin) [inline src/browser/window.ts:44]: Why do we need https here?

File: webpack.config.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ const isProd = process.env.NODE_ENV === 'production';
 const webpackConfig = {
   mode: isProd ? 'production' : 'development',
   devtool: 'eval-source-map',
-  target: 'electron-renderer',
+  target: 'web',
```
Comment (MasterOdin) [inline webpack.config.js:14]: Why the change away for electron as the target?

Comment (maxcnunes) [reply webpack.config.js:14]: The frontend code cannot access the electron module anymore, which turned the frontend more into a web app.

File: src/browser/window.ts (hunk 2 above)
Comment (maxcnunes) [reply src/browser/window.ts:44]: Just to avoid warning security errors in the chrome console during development

**Review by MasterOdin — APPROVED** [review]
Small comments, otherwise looks good, good job on this :+1:

File: src/browser/window.ts (hunk 2 above)
Comment (maxcnunes) [reply src/browser/window.ts]: @MasterOdin I just removed this change from this PR

File: src/browser/window.ts [hunk 4]
```diff
@@ -29,8 +29,10 @@ export function buildNewWindow(app: App): void {
     minWidth: 512,
     minHeight: 350,
     webPreferences: {
+      nodeIntegration: false, // is default value after Electron v5
+      contextIsolation: true, // protect against prototype pollution
+      enableRemoteModule: false, // turn off remote
```
[the remark this answers is not in the document]
Comment (maxcnunes) [reply src/browser/window.ts:34]: Alright. Since the upgrading has been done in that other PR, I am gonna remove these properties there.

File: src/renderer/actions/config.ts [hunk 5]
```diff
@@ -36,7 +32,8 @@ export function saveConfig(configData: Config): ThunkResult<void> {
     dispatch({ type: SAVE_CONFIG_REQUEST });
     try {
       await sqlectron.config.saveSettings(configData);
-      await sqlectron.setSelectLimit();
+      // TODO: Check, should it really be setting the limit wihtout a value?
+      sqlectron.db.setSelectLimit(1000);
```
[the remark this answers is not in the document]
Comment (maxcnunes) [reply src/renderer/actions/config.ts]: Done

File: src/browser/window.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (maxcnunes) [reply src/browser/window.ts:34]: done
